connector/connector.go

```diff
@@ -41,6 +44,12 @@ type Traces interface {
 	consumer.Traces
 }
 
+// TracesRouter feeds the first consumer.Traces in each of the specified pipelines.
+type TracesRouter interface {
+	consumer.Traces
+	RouteTraces(context.Context, ptrace.Traces, ...component.ID) error
```

Comment:
Feel like a better way is to give access to `map[PipelineID]consumer.Traces` (first consumer in each pipeline):
* Gives access to all pipelines: component knows how many and what are the IDs available, in this case the component has to be configured with the IDs. This is important for a load-balancer component where you want to do Round-Robin for example. 
* Avoid a slice creation because of the vararg (the compiler may be smart in some cases to stack allocate, but not sure).

Reply:
That's a good point about needing access to all pipelines, even when not sending a given request to all.

Do you agree that `map[PipelineID].consumer.Traces` fits better into https://github.com/open-telemetry/opentelemetry-collector/pull/7179, where this signature would be `Consumers() map[PipelineID]consumer.Traces`, rather than as a request to handle routing directly?

Reply:
Yes... Sorry will keep a bit the conversation here, or you can close this and copy the threads and continue there.

Reply:
No problem. Will keep discussion here for now at least.

Reply:
@bogdandrutu, I believe this one is what you suggested: https://github.com/open-telemetry/opentelemetry-collector/pull/7226